### PR TITLE
CSF workshop attendance export to gdrive

### PR DESCRIPTION
**Pull request by clareconstantine**
This will export a summary of CSF workshop data to a google sheet. I followed the process detailed in https://github.com/code-dot-org/code-dot-org/pull/32597.

## Testing story

Same process as https://github.com/code-dot-org/code-dot-org/pull/32597:

Tested end-to-end locally with development secrets configured in Secret Manager.

Note that the configuration added in this PR does not configure _development_ environments to pull anything from Secrets Manager; to do that for local testing, you have to add the following lines to your `config/develoment.yml.erb` file:

```yml
gdrive_export_secret: !Secret
applications_2020_2021_gsheet_key: !Secret
```

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: bin/cron/csf_workshop_attendance_to_gdrive [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+      ]
+    end
+
+  results
+end
+
+def scholarship_teacher_attendance(workshop)
+  num_teachers = 0
+  workshop.sessions.first.attendances.each do |attendance|
+    if Pd::ScholarshipInfo.where(pd_enrollment_id: attendance.pd_enrollment_id)
```
Comment (bencodeorg) [inline bin/cron/csf_workshop_attendance_to_gdrive]: Isn't "no" a valid scholarship status? Would this mark those teachers as scholarship recipients?

Comment (bencodeorg) [reply bin/cron/csf_workshop_attendance_to_gdrive]: We also have this pretty similar method in the workshop model that I think I added for something similar last year, I wonder if it's worth generalizing / making sure that the way "scholarship attendance" is being calculated here is consistent with how we're doing the same thing elsewhere?

https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/models/pd/workshop.rb#L687

File: bin/cron/csf_workshop_attendance_to_gdrive [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+  Pd::Workshop.
+    where(subject: Pd::SharedWorkshopConstants::SUBJECT_CSF_101).
+    scheduled_end_on_or_after(EARLIEST_WORKSHOP_END_DATE).
+    scheduled_start_on_or_before(LATEST_WORKSHOP_START_DATE).find_each do |workshop|
+      results << [
+        workshop.organizer&.name,
+        workshop.organizer&.email,
+        workshop.regional_partner&.name,
+        workshop.sessions.map(&:formatted_date_with_start_and_end_times).join("\n"),
```
Comment (bencodeorg) [inline bin/cron/csf_workshop_attendance_to_gdrive:45]: We want new lines in the gsheet export?

**Comment by bencodeorg** [comment]
General question: how does what is done here differ from what is produced in the various serializers we have for exporting workshop data?

File: bin/cron/csf_workshop_attendance_to_gdrive (hunk 1 above)
Comment (clareconstantine) [reply bin/cron/csf_workshop_attendance_to_gdrive]: oh excellent point - I totally forgot "No" is a status! And thanks for pointing me to this method! This export was initially getting attendance for each session, but now that I don't need to do that, I can reuse the method from the model. Thanks!

File: bin/cron/csf_workshop_attendance_to_gdrive (hunk 2 above)
Comment (clareconstantine) [reply bin/cron/csf_workshop_attendance_to_gdrive:45]: It creates a list of the workshop sessions:
Screen Shot 2021-02-21 at 1 04 15 PM

File: cookbooks/cdo-apps/templates/default/crontab.erb [hunk 3]
```diff
@@ -94,6 +94,8 @@
       cronjob at:'* * * * *', do:deploy_dir('bin', 'cron', 'confirm_usage')
       cronjob at:'0 */2 * * *', do:deploy_dir('bin', 'cron', 'teacher_applications_to_gdrive')
       cronjob at:'30 */2 * * *', do:deploy_dir('bin', 'cron', 'summer_workshops_to_gdrive')
+      # [Clare] Disabled so that we can do the first run manually on production-daemon, to update credentials.
```
Comment (cforkish) [inline cookbooks/cdo-apps/templates/default/crontab.erb:97]: can you elaborate on this? is the plan to re-enable it in a followup PR?

**Review by cforkish — APPROVED** [review]
LGTM!

File: cookbooks/cdo-apps/templates/default/crontab.erb (hunk 3 above)
Comment (clareconstantine) [reply cookbooks/cdo-apps/templates/default/crontab.erb:97]: Yes! I will run it manually the first time, and if all goes well, I'll submit a follow up to enable this.

**Pull request merged**